src/main/java/com/gluonhq/richtext/RichTextAreaSkin.java

```diff
@@ -305,6 +335,31 @@ private void updateLayers(SetChangeListener.Change<? extends Path> change) {
         }
     }
 
+    private void updateSelection(Selection selection) {
+        selectionShape.getElements().clear();
+        if (selection.isDefined()) {
+            selectionShape.getElements().setAll(textFlow.rangeShape(selection.getStart(), selection.getEnd()));
+        }
+        selectionShape.setLayoutX(textFlowLayoutX);
+        selectionShape.setLayoutY(textFlowLayoutY);
+    }
+
+    private void updateCaretPosition(int caretPosition) {
+        caretShape.getElements().clear();
+        if (caretPosition < 0) {
+            caretTimeline.stop();
+        } else {
+            var pathElements = textFlow.caretShape(caretPosition, true);
+            caretShape.getElements().addAll(pathElements);
+            if (caretShape.getLayoutBounds().getHeight() < 3) {
+                caretShape.getElements().add(new LineTo(0, 20));
```

Comment:
What's the significance of these magic numbers: 3 and 20 ?

Reply:
This was preventing a too tiny caret when the TextBuffer was empty and the initial line height was too short. 
However, after my last commit, seems this is no necessary (the caret now gets a height of 21.54). I'll remove this.